tests/integration/test_ingress.py

```diff
@@ -32,3 +38,35 @@ async def test_ingress_integration(
         timeout=5,
     )
     assert response.status_code == 200
+
+
+@pytest.mark.abort_on_fail
+async def test_agent_discovery_ingress_integration(
+    model: Model,
+    application: Application,
+    traefik_application: Application,
+    external_hostname: str,
+    jenkins_k8s_agents: Application,
+):
+    """
+    arrange: deploy the Jenkins charm and establish relations via ingress.
+    act: send a request to the ingress in /.
+    assert: the response succeeds.
```

Comment:
Does the test really do this? I actually see more of a relational data check.

Reply:
It doesn't actually, and the test was not very stable so I updated the test and this docstring.